lib/modules/dosomething/dosomething_reportback/includes/dosomething_reportback_file.inc

```diff
@@ -169,6 +169,26 @@ class ReportbackFileEntityController extends EntityAPIController {
   }
 
   /**
+   * Overrides save() method.
+   *
+   * Updates count variables for corresponding nodes and taxonomy terms.
+   */
+  public function save($entity, DatabaseTransaction $transaction = NULL) {
+    if (DOSOMETHING_REPORTBACK_LOG) {
+      watchdog('dosomething_reportback_file', 'save:' . json_encode($entity));
+    }
+    $status = NULL;
```

Comment:
I think it would be better to move this and the next `if` statement below `parent::save()` function. Nothing here is set that is affecting, `parent::save()`, but related to the count reset.